### PR TITLE
Upgrade vite and other improvements

### PR DESCRIPTION
**Pull request by yanfali**
Upgrades to npm packages and other improvements

 - adds husky to run pre-commit hooks to make contributing easier
 - upgrades @cypress/vue
 - upgrades vite
 - adds pprettier to speed up prettier checks
 - removes unused packages
 - upgrades node-sass

### CODE REVIEW

**Comment by yanfali** [comment]
@zekth quick look please

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Comment by yanfali** [comment]
> Looks good to me.

👍

**Review by fauxpark — APPROVED** [review]

**Review by zekth — APPROVED** [review]
:v:

File: package.json [hunk 1]
```diff
@@ -58,15 +59,15 @@
     "eslint-plugin-vue": "8",
     "growl": "^1.10.0",
     "https-proxy-agent": "^2.2.0",
-    "jest": "24",
```
Comment (mateossh) [inline package.json:61]: I am not sure, but if you remove `jest` package, then shouldn't you also remove `babel-jest`?

**Pull request merged**